src/libraries/Microsoft.PowerFx.Connectors/Public/ConnectorType.cs

```diff
@@ -20,7 +20,7 @@ namespace Microsoft.PowerFx.Connectors
     // FormulaType is used to represent the type of the parameter in the Power Fx expression as used in Power Apps
     // ConnectorType contains more details information coming from the swagger file and extensions
     [DebuggerDisplay("{FormulaType._type}")]
-    public class ConnectorType : SupportsConnectorErrors
+    public class ConnectorType : SupportsConnectorErrors, IEquatable<ConnectorType>
```

Comment:
>IEquatable<ConnectorType> [](http://example.com/codeflow?start=58&length=25)

Why do we need this?

Reply:
IEquatable<T> is the interface implementing Equals(T)
It's a good practice to use this interface when you implement Equals

See [https://learn.microsoft.com/en-us/dotnet/api/system.iequatable-1?view=net-8.0](https://learn.microsoft.com/en-us/dotnet/api/system.iequatable-1?view=net-8.0)

The IEquatable<T> interface is used by generic collection objects such as Dictionary<TKey,TValue>, List<T>, and LinkedList<T> when testing for equality in such methods as Contains, IndexOf, LastIndexOf, and Remove. It should be implemented for any object that might be stored in a generic collection.

Reply:
but why are we putting these in a collection? My concern is the complexity of .equals below. Is there some simplification we can apply here?   Ie, is it sufficient to just compare logical names; or do we need a full comparison?

Reply:
We can probably simplify it a bit (logical names aren't enough) and i'll check
We need it due to the change in CompileTimeTypeWrapperRecordValue
IEquatable<T> is a best practice anyway, regardless of using collections

Reply:
A ConnectorType is a sorta converted version of a Swagger schema and here we are comparing all elements we can identify in the schema ... and there are simply a lot
We could imagine converting the schema to a string and make a comparison of that string but the equality won't be valid in all cases as you could define types locally or as a reference, both identical, but get an equality difference (this is just one example)